shared/src/balancer/event_handler.rs

```diff
@@ -285,15 +426,32 @@ impl_event_retrieving! {
     pub BalancerV2WeightedPoolFactoryContract for balancer_v2_weighted_pool_factory
 }
 
+impl_event_retrieving! {
+    pub BalancerV2WeightedPool2TokensFactoryContract for balancer_v2_weighted_pool_2_tokens_factory
+}
+
 #[async_trait::async_trait]
 impl Maintaining for BalancerEventUpdater {
     async fn run_maintenance(&self) -> Result<()> {
-        self.0.run_maintenance().await
+        self.two_token_pool_updater.run_maintenance().await?;
+        self.weighted_pool_updater.run_maintenance().await
```

Comment:
This is responsible for deleting all the original pools and starting over again. Need to separate last updated in a way that it knows not to delete the other pools.